addon/modifiers/did-intersect.js

```diff
@@ -38,15 +47,38 @@ export default class DidIntersectModifier extends Modifier {
       return;
     }
 
-    let { onEnter, onExit, options } = this.args.named;
+    let { onEnter, onExit, maxEnter, maxExit, options } = this.args.named;
 
     assert('onEnter or/and onExit is required', onEnter || onExit);
 
-    if (onEnter) {
-      this.observerManager.addEnterCallback(this.element, onEnter);
+    // We should technically accept 0, in case of programmatic changes
+    if (maxEnter || maxEnter === 0) {
+      assert('maxEnter must be an integer', Number.isInteger(maxEnter));
+      this._maxEnterIntersections = maxEnter;
     }
-    if (onExit) {
-      this.observerManager.addExitCallback(this.element, onExit);
+
+    // We should accept 0 here too.
+    if (maxExit || maxExit === 0) {
+      assert('maxExit must be an integer', Number.isInteger(maxExit));
+      this._maxExitIntersections = maxExit;
+    }
+
+    if (onEnter && maxEnter !== 0) {
+      this.observerManager.addEnterCallback(this.element, (...args) => {
+        if (!maxEnter || this._numOfEnters < maxEnter) {
```

Comment:
You want to reference `this._maxEnterIntersections` here, and the same below with Exit

Reply:
done